### ISSUE TITLE
Issue

### ISSUE THREAD

**Issue by bloopbytes** [davidmhewitt/clipped#30]
I can open only once, then doesn't work, must logout and login again

**Comment by NathanBnm** [comment]
I am the same issue. When I open Clipped I can use it, then it crashes and I can't reopen it without ending the process or logout and login again.

**Comment by davidmhewitt** [comment]
Thanks for the report, I can't reproduce this. Can you please give more details that may help?

For example, what shortcut have you chosen? What keyboard layout are you using? Does clipped open if you choose it from the applications menu instead of using the keyboard shortcut?

**Comment by NathanBnm** [comment]
I'll check that asap and will tell you

**Comment by NathanBnm** [comment]
So I have the Ctrl + Alt + V shortcut and I'm using the French (AZERTY) keyboard layout.

Clipped opens when I use the shortcut but also if I use the launcher.

The difference is that when I open it from the Applications menu it appears minimized and I have to click on it on the dock to make it appear on the screen (with the shortcut it appears directly).

But that seems to be broken at the moment I lost the focus on the window. I mean when I right click (or right) click on any other element (clicking on the dock, even right clicking on the Clipped icon or clicking the elementary topbar, or switch to another app).

At this moment Clipped just closes (or crash idk), and then I can't reopen it, both with shortcut and launcher.

Is that more clear for you?

**Comment by davidmhewitt** [comment]
Ok, great, thanks. I still can't reproduce that particular issue with the combination of that shortcut and keyboard layout. The window is supposed to close when it loses focus, but re-opens normally when I use the shortcut again.

I think the issue with it opening weirdly from the applications menu is a wingpanel bug as it opens fine from the dock if you pin it there.

Would you mind trying compiling and installing from the `remove-x-keybind` branch (https://github.com/davidmhewitt/clipped/tree/remove-x-keybind) and seeing if that improves anything? Remember to kill the old running one first before trying :slightly_smiling_face:

**Comment by NathanBnm** [comment]
> Ok, great, thanks. I still can't reproduce that particular issue with the combination of that shortcut and keyboard layout. The window is supposed to close when it loses focus, but re-opens normally when I use the shortcut again.

Sure this should be the correct behaviour :wink:
> 
> I think the issue with it opening weirdly from the applications menu is a wingpanel bug as it opens fine from the dock if you pin it there.

Yeah maybe
> 
> Would you mind trying compiling and installing from the `remove-x-keybind` branch (https://github.com/davidmhewitt/clipped/tree/remove-x-keybind) and seeing if that improves anything? Remember to kill the old running one first before trying slightly_smiling_face

I'll try that

**Comment by NathanBnm** [comment]
Nope! I still have the same issue :confused:

**Comment by davidmhewitt** [comment]
Would you mind restarting after installing the one from that branch? Just in case there any old X11 keybindings left behind in memory from the old version. I can't understand this issue at all, I've set up a new user in French language with french keyboard, used the same shortcut as you and everything works fine :confused:

**Comment by NathanBnm** [comment]
I'm going to try again with that

**Comment by NathanBnm** [comment]
It's really hard to explain because it seems to be random crash

**Comment by NathanBnm** [comment]
I'll try to explain step by step:

First I open from the launcher and it opens the `com.github.davidmhewitt.clipped` process.
Then when I try to open it again it just crashes (it opens and closes immediately)

I kill the process

Now I open it with the Ctrl + Alt + V shortcut (the dock icon is blurred) and it opens the `com.github.davidmhewitt.clipped--show-paste-window` process.
Now I can't open it again using the shortcut (nothing appear).

Then if I open it from the launcher it opens but twice. Don't know how to explain. I'll make a little video capture to show you.

**Comment by NathanBnm** [comment]
You can watch my recording here (unlisted video) : https://youtu.be/6fI8ZVnkbIs

**Comment by davidmhewitt** [comment]
Interesting...

I've just pushed another branch called `wmclass`, could you compile and install that one and give it a try?

**Comment by davidmhewitt** [comment]
Also, could you open the Clipped preferences window and then run, `xprop WM_CLASS` in a terminal and then click the preferences window when you get the crosshair cursor. You should get some output in the terminal like:
`WM_CLASS(STRING) = "com.github.davidmhewitt.clipped", "Com.github.davidmhewitt.clipped"`

Can you let me know if that says something different to the above?

**Comment by NathanBnm** [comment]
Done, this is what I get right.

```
WM_CLASS(STRING) = "com.github.davidmhewitt.clipped", "Com.github.davidmhewitt.clipped"
```

**Comment by NathanBnm** [comment]
With this version, it's always working with the shortcut but it's still messy when I launch the app from the launcher

**Comment by NathanBnm** [comment]
I have no error when I try to launch it directly from command line, either with `com.github.davidmhewitt.clipped` and with `com.github.davidmhewitt.clipped --show-paste-window`

**Comment by davidmhewitt** [comment]
Thank you for all your testing, I think I've now solved two potential issues that may have been causing the shortcut key to not work properly.

Opening clipped from the launcher not working properly is a separate issue that I think is caused by wingpanel causing the clipped window to focus out on startup somehow, I believe this is now improved in `master`, but may still be a slight issue.

However, I don't think opening it from the menu is a very common use case, so hopefully the fixes to the shortcut will be more helpful.

Thank you again for all your help!

**Comment by NathanBnm** [comment]
That's great!

Sure about the launcher, but you should still try to fix it too I think

**Comment by davidmhewitt** [comment]
Version 1.1.1 has been released to the AppCenter and I believe this includes a fix for the shortcut key only working once.

As such, I'm going to close this issue, but please feel free to re-open it or add another comment if you're still experiencing this particular issue after updating.

**Comment by NathanBnm** [comment]
Sure I'll check that!
